### PR TITLE
Add component and element templates

### PR DESCRIPTION
**Pull request by JJRcop**
## About The Pull Request
Adds some templates for components and elements, which contain boilerplate code and some optional features commented out.

This can help create them faster, and may prevent some common mistakes.

### CODE REVIEW

**Comment by Mothblocks** [comment]
These should be in a `.md` file instead, tbqh, you'd get rid of a lot of the ugly stuff that way.

**Comment by JJRcop** [comment]
How do you mean, what would it look like?

**Comment by Mothblocks** [comment]
I would think just make a `.md` (or even just `.txt`) file thats wrapped in \`\`\`dm, you don't want people having to deal with your ifdefs and stuff.

CC @ninjanomnom on the templates themselves.

**Comment by JJRcop** [comment]
I agree with the `.md` file way of doing things

**Comment by JJRcop** [comment]
This pull request is complete, and I have no pending requests from maintainers.

File: code/datums/elements/ELEMENT_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+See _element.dm for detailed explanations
+
+```dm
+/datum/element/myelement
+	element_flags = ELEMENT_BESPOKE | ELEMENT_COMPLEX_DETACH | ELEMENT_DETACH | ELEMENT_NOTAREALFLAG    // code/__DEFINES/dcs/flags.dm
+	//id_arg_index = 2                                                                                  // Use with ELEMENT_BESPOKE
+	var/list/myvar = list()
+
+/datum/element/myelement/Attach(datum/target)
+	myvar |= target
```
Comment (ninjanomnom) [inline code/datums/elements/ELEMENT_TEMPLATE.md:13]: This sort of code should not be in the template as it's rare that an element should be keeping track of state. Having it in the template implies that it is generally acceptable which it isn't.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/elements/ELEMENT_TEMPLATE.md (hunk 1 above)
Comment (JJRcop) [reply code/datums/elements/ELEMENT_TEMPLATE.md:13]: I want some kind of example code here, however tiny. Maybe just a simple `target.name` change.

Comment (ninjanomnom) [reply code/datums/elements/ELEMENT_TEMPLATE.md]: Show something like playing a sound on move, just changing a var isn't really a good use of an element either.

**Comment by JJRcop** [comment]
I have addressed all reviews

**Comment by JJRcop** [comment]
@ninjanomnom

**Pull request merged**